### PR TITLE
[Poem Art] Add contest winner poem to English set

### PR DESCRIPTION
**Pull request by mikeharv**
This adds one new poem, listed in the selector only when the site is in English ~or English (UK) DBG~. 
It replaces the `locale` (string) attribute with the `locales` (array) attribute.

## Links

- jira ticket: [STAR-2302: [Poem Art] Add contest winner poem to English set [DUE 6/3]](https://codedotorg.atlassian.net/browse/STAR-2302)

## Testing story
We also have poems that are only displayed if the page is in Polish. I checked the selector and compared the list across both English locales, as well as Polish, and Spanish (for a control).

- Tested that "My City" shows in ~both~ English locale, but not otherwise.
- Tested that poems like "WIESZAKI" show correctly in Polish, but not otherwise.
- Tested that none of the hard-coded poems show in Spanish.

image
image
image

### CODE REVIEW

File: apps/src/p5lab/poetry/PoemSelector.jsx [hunk 1]
```diff
@@ -185,7 +185,11 @@ function PoemSelector(props) {
   const getPoemOptions = () => {
     const options = Object.keys(POEMS)
       .map(poemKey => getPoem(poemKey))
-      .filter(poem => !poem.locale || poem.locale === appOptions.locale)
+      .filter(
+        poem =>
+          !poem.locales ||
+          poem.locales.some(locale => locale === appOptions.locale)
```
Comment (breville) [inline apps/src/p5lab/poetry/PoemSelector.jsx]: Could you just use `.includes`?

File: apps/src/p5lab/poetry/constants.js [hunk 2]
```diff
@@ -28,10 +28,10 @@ export const PALETTES = {
 
 // Notes:
 // - author is not translated.
-// - Poems are shown in all languages, unless there is a locale attribute, in which case
-//   the poem is shown in the dropdown only for users with that current locale.
-// - If the locale attribute is set, then title is used and is not translated.
-// - If the locale attribute is set, then linesSplit is used and is not translated.
+// - Poems are shown in all languages, unless there is a locales attribute, in which case
+//   the poem is shown in the dropdown only for users with a current locale that matches.
```
Comment (breville) [inline apps/src/p5lab/poetry/constants.js]: super nit: "that matches" is slightly ambiguous, perhaps "that is included in the array"?

File: apps/src/p5lab/poetry/PoemSelector.jsx (hunk 1 above)
Comment (mikeharv) [reply apps/src/p5lab/poetry/PoemSelector.jsx]: For sure! Much simpler. :)

File: apps/src/p5lab/poetry/constants.js (hunk 2 above)
Comment (mikeharv) [reply apps/src/p5lab/poetry/constants.js]: Updated!

**Review by breville — APPROVED** [review]
Thanks for testing all the different cases, too.

**Comment by mikeharv** [comment]
@KylieModen This is basically ready to go. The last confirmation I need from you is whether or not we are okaying with showing this poem to users who select "English (UK)" using the dropdown at the bottom: 
image
Currently this is set to show for both "English" and "English (UK)". Alternative options include showing the poem for just "English", but not "English (UK)", or possibly including other locales.

**Comment by mikeharv** [comment]
@KylieModen Ignore previous comment! Brendan pointed out that English (UK) is not actually a choice on production. It shows up on localhost, likely just for debugging purposes.

**Review by KylieModen — COMMENTED** [review]
Do we know if English in the language dropdown corresponds to en-US? or could it correspond to generic en? 

image
^from https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/config/locales.yml

**Comment by mikeharv** [comment]
> Do we know if English in the language dropdown corresponds to en-US? or could it correspond to generic en?
> 
> image

@KylieModen If I add a `console.log(appOptions.locale)` to the selector, it prints `'en_us'` if the site is in English. It doesn't look like it would ever be just `'en'`, and if I try setting the poem's `locales` attribute to `['en']`, the poem isn't listed for any language.

**Comment by KylieModen** [comment]
In that case LGTM!

**Review by KylieModen — APPROVED** [review]

**Pull request merged**